lib/rust/parser/debug/src/lib.rs

```diff
@@ -199,37 +201,39 @@ pub fn validate_spans(
     tree: &enso_parser::syntax::tree::Tree,
     expected_span: std::ops::Range<u32>,
     locations: &mut LocationCheck,
-) {
+) -> Result<(), String> {
     let mut sum_span = None;
     fn concat<T: PartialEq + std::fmt::Debug + Copy>(
         a: &Option<std::ops::Range<T>>,
         b: &std::ops::Range<T>,
-    ) -> std::ops::Range<T> {
-        match a {
+    ) -> Result<std::ops::Range<T>, String> {
+        Ok(match a {
             Some(a) => {
-                assert_eq!(a.end, b.start);
+                if a.end != b.start {
+                    return Err(format!("{:?} != {:?}", &a.end, b.start));
+                }
                 a.start..b.end
             }
             None => b.clone(),
-        }
+        })
     }
-    sum_span = Some(concat(&sum_span, &tree.span.left_offset.code.range()));
+    sum_span = Some(concat(&sum_span, &tree.span.left_offset.code.range())?);
     tree.visit_items(|item| match item {
         enso_parser::syntax::item::Ref::Token(token) => {
             if !(token.left_offset.is_empty() && token.code.is_empty()) {
-                sum_span = Some(concat(&sum_span, &token.left_offset.code.range()));
-                sum_span = Some(concat(&sum_span, &token.code.range()));
+                sum_span = Some(concat(&sum_span, &token.left_offset.code.range()).unwrap());
+                sum_span = Some(concat(&sum_span, &token.code.range()).unwrap());
```

Comment:
The `unwrap`s are used as debug assertions. I'm moving toward `Result` so that when failure occurs, context information can be attached; the remaining cases aren't simple to migrate because there are lambdas here that will need to exfiltrate errors.